fMRIVolume/GenericfMRIVolumeProcessingPipeline.sh

```diff
@@ -293,15 +457,82 @@ if [ ! -e "$fMRIFolder" ] ; then
   log_Msg "mkdir ${fMRIFolder}"
   mkdir "$fMRIFolder"
 fi
-cp "$fMRITimeSeries" "$fMRIFolder"/"$OrigTCSName".nii.gz
+${FSLDIR}/bin/imcp "$fMRITimeSeries" "$fMRIFolder"/"$OrigTCSName"
+
+# --- Do slice time correction if indicated
+if [ $DoSliceTimeCorrection = "TRUE" ] ; then
+    log_Msg "Running slice timing correction using FSL's 'slicetimer' tool"
+    TR=`${FSLDIR}/bin/fslval "$fMRIFolder"/"$OrigTCSName" pixdim4`
+    log_Msg "TR: ${TR}"
+
+    IFS='@' read -a SliceTimerCorrectionParametersArray <<< "$SliceTimerCorrectionParameters"
+    ${FSLDIR}/bin/immv "$fMRIFolder"/"$OrigTCSName" "$fMRIFolder"/"$OrigTCSName"_prestc
+    ${FSLDIR}/bin/slicetimer -i "$fMRIFolder"/"$OrigTCSName"_prestc -o "$fMRIFolder"/"$OrigTCSName" -r ${TR} -v "${SliceTimerCorrectionParametersArray[@]}"
+    ${FSLDIR}/bin/imrm "$fMRIFolder"/"$OrigTCSName"_prestc
+fi
 
-#Create fake "Scout" if it doesn't exist
-if [ $fMRIScout = "NONE" ] ; then
-  ${RUN} ${FSLDIR}/bin/fslroi "$fMRIFolder"/"$OrigTCSName" "$fMRIFolder"/"$OrigScoutName" 0 1
-else
-  cp "$fMRIScout" "$fMRIFolder"/"$OrigScoutName".nii.gz
+# --- Copy over scout (own or reference if specified), create fake if none exists
+
+if [ "$fMRIReference" != "NONE" ]; then
+    # --- copy over exisiting scout images
+    log_Msg "Copying Scout from Reference BOLD"
+    ${FSLDIR}/bin/imcp ${fMRIReferencePath}/Scout* ${fMRIFolder}
+
+    for simage in SBRef_nonlin SBRef_nonlin_norm
+    do
+        ${FSLDIR}/bin/imcp ${fMRIReferencePath}/"${fMRIReference}_${simage}" ${fMRIFolder}/"${NameOffMRI}_${simage}"
+    done
+
+    mkdir -p ${ResultsFolder}
+    ${FSLDIR}/bin/imcp ${ReferenceResultsFolder}/"${fMRIReference}_SBRef" ${ResultsFolder}/"${NameOffMRI}_SBRef"
```

Comment:
Were we wanting to use symlinks rather than copies here, so that there is an indication of where the data came from?

Reply:
As I wrote before, in our experience symlinks can be an issue in interaction with XNAT uptake. Also, if for any reason the reference BOLD is deleted then the referenced image is lost. And if the reference BOLD is redone the symlink will be misleading. To be on the safe side, I would use copy and not symlinks.

Reply:
I don't think we've run into a problem in using symlinks in other scripts, but ok.

Reply:
Symlinking across different BOLD directories doesn't sound like a good idea to me, too many potential surprises when reprocessing one run and not another.  The other symlinks I know of in the pipelines don't cross between runs.